dht/i_partitioner.hh

```diff
@@ -317,6 +318,9 @@ class ring_position_view {
     const dht::token* _token; // always not nullptr
     const partition_key* _key; // Can be nullptr
     int8_t _weight;
+private:
+    ring_position_view() noexcept : _token(nullptr), _key(nullptr), _weight(0) { }
+    operator bool() const noexcept { return bool(_token); }
 public:
```

Comment:
Good that you placed this in private namespace, as other users of ring_position_view may be relying on token always not null.

Reply:
nit: we usually define `operator bool()` as explicit
do we need this one to be implicit?

Reply:
I knew I forgot to add some decorator I just couldn't remember which. Will add.